### PR TITLE
Adding Object Recognition Source

### PR DESCRIPTION
**Pull request by AlexBlumer**
A customizable, extensible source intended for object recognition.

The edu.ml source can largely be ignored. It is someone else's YOLO implementation in Java that is included here in case the source changes, and is largely unedited except to give us access to the data how we want it.

### CODE REVIEW

File: objectRecognitionSource/src/dist/scripts/replaceJni [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#Save relevant file locations
```
[the remark this answers is not in the document]
Comment (AlexBlumer) [reply objectRecognitionSource/src/dist/scripts/replaceJni]: These are not generated BUT they were written for the openpnp distribution of OpenCV, which we're no longer using. I'll remove them

File: objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/CameraRetriever.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	/**
+	 * Obtain the most recent image from the camera
+	 * @throws FatalImageException     If the camera is no longer open.
+	 */
+	@Override
+	public ImageRetrieverResults getImage() throws ImageAcquisitionException {
+		// Read the next video frame from the camera
+		Mat matrix = new Mat();
+		ImageRetrieverResults results = new ImageRetrieverResults();
+        Date captureTime = new Date();
```
[the remark this answers is not in the document]
Comment (AlexBlumer) [reply objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/CameraRetriever.java]: Yep, a lot of the indents were actual tabs. I just did a mass replace for spaces.

**Comment by fhcarter** [comment]
@AlexBlumer Will do (the gradlew change).

That said, the ftp tests are now failing because 'test.rebex.net' is not a valid URL.  So, your `stripDomainName()` was a bit more general than the URL version.  If you want to revert the change but change the name to `extractDomainName` or something, that's OK.

**Comment by AlexBlumer** [comment]
I looked again and apparently the replaceJni files were for custom built tensorflow, not openpnp. Should I add them back in or let the user figure out how to make it work? (Not 100% certain that they'll do enough to run the custom built version)

File: objectRecognitionSource/README.md [hunk 3]
```diff
@@ -389,19 +389,19 @@ No additional data is sent.
...
 build TensorFlow Java](https://www.tensorflow.org/install/install_sources#build_the_c_or_java_libraries). Once built,
-either replace the `libtensorflow-<version>.jar` in `<install location>/objectRecognitionSource/lib` and run
-`<install location>/objectRecognitionSource/scripts/replaceJni <the built .so/.dll file>`; or add the locations of the
-new files into the `CLASSPATH` variable found in `<install location>/objectRecognitionSource/bin/objectRecognitionSource`
+set the environment variable `TENSORFLOW_JNI` to the folder containing all the .dll/.so/.dylib files, and either replace
+the `libtensorflow-<version>.jar` in `<install location>/objectRecognitionSource/lib` with the built jar or add the jar
+into the `CLASSPATH` variable found in `<install location>/objectRecognitionSource/bin/objectRecognitionSource`
 and `<install location>/objectRecognitionSource/bin/objectRecognitionSource.bat` and remove the `libtensorflow*` files
-in `<install location>/objectRecognitionSource/lib`.  
+in `<install location>/objectRecognitionSource/lib`. (This method is expected, but not certain not work).
```
Comment (fhcarter) [inline objectRecognitionSource/README.md]: Please remove the disclaimer.  If it doesn't work, they can file a bug.

You should be able to test this simply by moving the `libtensorflow-<version>.jar` file to somewhere else and making the same changes.  And/or renaming it (maybe).

Comment (AlexBlumer) [reply objectRecognitionSource/README.md]: Works for moving the jni, and classpath mostly just depends on the users

**Pull request merged**